src/posterior_gp/approx_posterior_gp.jl

```diff
@@ -22,15 +22,95 @@ function approx_posterior(::VFE, fx::FiniteGP, y::AbstractVector{<:Real}, u::Fin
     U = cholesky(Symmetric(cov(u))).U
     B_εf = U' \ (U_y' \ cov(fx, u))'
     b_y = U_y' \ (y - mean(fx))
-    Λ_ε = cholesky(Symmetric(B_εf * B_εf' + I))
+    D = B_εf * B_εf' + I
+    Λ_ε = cholesky(Symmetric(D))
     m_ε = Λ_ε \ (B_εf * b_y)
-    return ApproxPosteriorGP(VFE(), fx.f, (m_ε=m_ε, Λ_ε=Λ_ε, U=U, α=U \ m_ε, z=u.x))
+    return ApproxPosteriorGP(VFE(), fx.f,
+                             (m_ε=m_ε,
+                              Λ_ε=Λ_ε,
+                              U=U,
+                              α=U \ m_ε,
+                              z=u.x,
+                              b_y=b_y,
+                              B_εf=B_εf,
+                              D=D,
+                              x=fx.x,
+                              Σy=fx.Σy,
+                              ))
```

Comment:
Style. I would suggest
```suggestion
    cache = (
        m_ε=m_ε,
        Λ_ε=Λ_ε,
        U=U,
        α=U \ m_ε,
        z=u.x,
        b_y=b_y,
        B_εf=B_εf,
        D=D,
        x=fx.x,
        Σy=fx.Σy,
    )
    return ApproxPosteriorGP(VFE(), fx.f, cache)
```